File: python/36.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        # check rows
        for row in board:
            if not self.check_valid(row):
                return False
        
        # check columns
        for col in range(9):
            column = [board[row][col] for row in range(9)]
            if not self.check_valid(column):
                return False
        
        # check sub-boxes
        for i in range(0, 9, 3):
            for j in range(0, 9, 3):
                sub_box = [board[row][col] for row in range(i, i+3) for col in range(j, j+3)]
                if not self.check_valid(sub_box):
                    return False
        
        return True
    
    def check_valid(self, lst: List[str]) -> bool:
        nums = [n for n in lst if n != '.']
        return len(set(nums)) == len(nums)

File: C#/36.cs
public class Solution {
    public bool IsValidSudoku(char[][] board) {
        // Check each row
        for (int i = 0; i < 9; i++) {
            HashSet<char> seen = new HashSet<char>();
            for (int j = 0; j < 9; j++) {
                char c = board[i][j];
                if (c != '.') {
                    if (seen.Contains(c)) {
                        return false;
                    }
                    seen.Add(c);
                }
            }
        }
        // Check each column
        for (int j = 0; j < 9; j++) {
            HashSet<char> seen = new HashSet<char>();
            for (int i = 0; i < 9; i++) {
                char c = board[i][j];
                if (c != '.') {
                    if (seen.Contains(c)) {
                        return false;
                    }
                    seen.Add(c);
                }
            }
        }
        // Check each 3x3 sub-box
        for (int box = 0; box < 9; box++) {
            HashSet<char> seen = new HashSet<char>();
            for (int i = box / 3 * 3; i < box / 3 *